app/controllers/home_controller.rb

```diff
@@ -75,6 +77,24 @@ def pii_phi_information
 
   private
 
+  def practice_names_and_slugs_hash
+    practices_and_slugs = {}
+    if current_user
+      all_current_practices.pluck("name", "slug").map { |name, slug| practices_and_slugs[name] = slug }
+    else
+      all_public_practices.pluck("name", "slug").map { |name, slug| practices_and_slugs[name] = slug }
+    end
+    return practices_and_slugs
+  end
+
+  def all_current_practices
+    Practice.where(published: true, retired: false).order("created_at DESC")
+  end
+
+  def all_public_practices
+    Practice.where(published: true, is_public: true, retired: false).order("created_at DESC")
+  end
```

Comment:
this works well, but at the risk of getting a bit code-golf-y we could potentially DRY things up just a bit:

```suggestion
 def practice_names_and_slugs_hash
  practices = current_user ? all_practices(current: true) : all_practices(public: true)
  practices.pluck("name", "slug").each_with_object({}) do |(name, slug), hash|
    hash[name] = slug
  end
end

def all_practices(current: false, public: false)
  scope = Practice.where(published: true, retired: false)
  scope = scope.where(is_public: true) if public
  scope = scope.order("created_at DESC")
  scope
end
```

since we are not utilizing the return value of the `map`s, I think it's best practice to use `each`, but then we can take it just a step further and use `each_with_object` with a hash to reduce repeated code and save a few lines. 

Not super necessary if you feel strongly that yours reads better 👍🏼